src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarNationEventListener.java

```diff
@@ -243,19 +225,20 @@ public void onNationStatusScreen(NationStatusScreenEvent event) {
 	}
 	
 	/*
-	 * A nation being deleted with a siege means the siege ends.
+	 * A nation being deleted with a siege means the siege ends,
+	 * and a king may receive a refund.
 	 */
-	@EventHandler
-	public void onDeleteNation(DeleteNationEvent event) {
-		UUID kingUUID = event.getNationKing();
-		if (kingUUID == null)
-			return;
-		
-		Resident king = TownyUniverse.getInstance().getResident(kingUUID);
-		if (king != null)
-			SiegeWarMoneyUtil.makeNationRefundAvailable(king);
+	@EventHandler(priority=EventPriority.MONITOR, ignoreCancelled=true)
+	public void onDeleteNation(PreDeleteNationEvent event) {
+		//If nation refund is enabled, warn the player that they will get a refund (and indicate how to claim it,) and then make it available.
+		if (TownyEconomyHandler.isActive() && SiegeWarSettings.getWarSiegeRefundInitialNationCostOnDelete() && event.getNation().getKing() != null) {

```

Comment:
It might be worth adding a check to make sure the refund amount percentage is above zero.

Reply:
Done, as well as removing doubled-up testing later in the SiegeWarMoneyUtil method.